pkg/controllers/management/auth/globalrolebinding_handler.go

```diff
@@ -149,8 +148,16 @@ func (grb *globalRoleBindingLifecycle) reconcileGlobalRoleBinding(globalRoleBind
 	}
 	globalRoleBinding.Annotations[crbNameAnnotation] = crbName
 
+	return grb.addRulesForTemplateAndTemplateVersions(globalRoleBinding, subject)
+}
+
+func (grb *globalRoleBindingLifecycle) addRulesForTemplateAndTemplateVersions(globalRoleBinding *v3.GlobalRoleBinding, subject v1.Subject) error {
+	var catalogTemplateRule, catalogTemplateVersionRule *v1.PolicyRule
```

Comment:
In this method, for lines 162-179, in the for loop, we create only one catalogTemplateRule and catalogTemplateVersionRule. Can there be more than one templateResourceRule or templateVersionResourceRule? If yes then catalogTemplateRule and catalogTemplateVersionRule will get overwritten.

Reply:
@prachidamle I don't think it's common to create role with more than one rules for the same resource. Because each rule needs verbs for that resource, so I don't know what happens if they give one rule for templates with verbs "*" and one rule for "get".

Reply:
@mrajashree  Whether to add all verbs in one rule or in a separate rule, that is upto the user of the API. Since API does not prevent having multiple rules for same resource, logic handling the rules should not assume that there will be single rule.